### PR TITLE
Airport Challenge - Elle Ordoña

### PR DESCRIPTION
**Pull request by elleordona**
airport challenge completed by Elle Ordoña

### CODE REVIEW

**Review by elleordona — COMMENTED** [review]
clearly got bored but  nice, should've really done the last part of the extension criteria

**Review by elleordona — COMMENTED** [review]
downloaded a formatting extension so most changes are just formatting.
main change was fixing the nesting of landPlane and takeoffPlane when implementing weather

**Review by elleordona — COMMENTED** [review]
good awareness that a part of the code was not in use and got rid of it

**Review by elleordona — COMMENTED** [review]
use a better commit message
changed the toggleWeather function so that is easier to declare if weather is to be considered when landing and taking off planes. found this was a problem in the previous commit and changed it

**Review by elleordona — COMMENTED** [review]
found you needed to toggleWeather before every function if you wanted to implement it.
thats a bit long but it got fixed so yay

**Review by elleordona — COMMENTED** [review]
fixed the original solution to pass test F-I but it was still nested :/ no bueno

**Review by elleordona — COMMENTED** [review]
the length of takeoffPlane is embarrassing ;_; 
thank god i fixed this eventually
honestly i shouldnt have accepted this solution in the first place but i remember it was like 1am when i committed this so i will allow it i guess

**Review by elleordona — COMMENTED** [review]
never used this test
i actually dont remember when i got rid of it

**Review by elleordona — COMMENTED** [review]
this got changed later but it was a nice thought

**Review by elleordona — COMMENTED** [review]
idk what happened here??

**Review by elleordona — COMMENTED** [review]
started trying to decouple Airport and Plane 👍 , technically week two stuff but good practice.
had to fix some of the first tests so that they would pass with the new code which i think is ok??

**Review by elleordona — COMMENTED** [review]
installed eslint for formatting and a lot of small things changed
all good tho cus it didnt mess with the function of the code
also this changed all my arrow functions into normal ones but i think i like them better? 🤔

**Review by elleordona — COMMENTED** [review]
made airportRunner a bit clearer when it was run
also added an extra part for displaying the planes at an airport which was unnecessary so i could have just left it out

**Review by elleordona — COMMENTED** [review]
completed main criteria so good to look at the extended stuff

**Review by elleordona — COMMENTED** [review]
committed this during the lesson but then Ed said "dont use static method in airport challenge" so this idea got scrapped v quickly

**Review by elleordona — COMMENTED** [review]
these edge cases got scrapped after a conversation with Alistar.
they kind of went further that what was expected and what I had at this time would have worked for the main criteria

**Review by elleordona — COMMENTED** [review]
found a different edge case 👍 
which got sorted in later commits

**Review by elleordona — COMMENTED** [review]
created a runner to test airport functions
started thinking about decoupling 👀

**Review by elleordona — COMMENTED** [review]
realised i never cleaned up the plane variable after each test so i fixed that.
at the time i forgot that i did used input as a variable in a couple tests but i also added that to the clean up in a later commit

**Review by elleordona — COMMENTED** [review]
god bless Ed for telling me that we cant use === to compare two objects 🙌 
this stumped me lol

**Review by elleordona — COMMENTED** [review]
all the commits from this week went by flawlessly
honestly i just wished i knew about jasmine so i didnt have to do my spec this really long way but it was really useful at the time for understanding TDD
now i love TDD makes my life so much easier ❤️

**Review by elleordona — COMMENTED** [review]
this got scrapped after we learnt about decoupling lol
knowing what i know now BAD PRACTICE DONT EVER HAVE THIS STEP AGAIN 😂

**Review by elleordona — COMMENTED** [review]
good to realise i needed to make sure it worked both ways
also figured out that the output of a boolean was not a string like the output of typeof

**Comment by elleordona** [comment]
honestly great project but i just clocked 99 commits!! i didnt even finish the extended criteria
idk if thats too much but 🤷 it helped me understand what happened at each step
LMK if its a bit much

File: .eslintrc.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ module.exports = {
     },
     'extends': 'eslint:recommended',
     'parserOptions': {
-        'ecmaVersion': 12
+        'ecmaVersion': 2020
```
Comment (elleordona) [inline .eslintrc.js:9]: tried to fix a formatting error but i just ended up disabling it for that document 😂
